src/inmanta/agent/executor.py

```diff
@@ -333,16 +333,16 @@ def __init__(self, envs_dir: str) -> None:
         self._cleanup_scheduler = util.Scheduler("venv_cleanup_scheduler")
 
     async def start(self) -> None:
-        interval = datetime.timedelta(days=1).total_seconds()
+        interval_cleanup_check = datetime.timedelta(days=1).total_seconds()
```

Comment:
Why check only once per day? 

I would expect we support retention times that are much shorter? We write the keep alive every minute, so we can go as low as e.g. 61 seconds here? 



I'm also going to point to @sanderr remark here: 

https://github.com/inmanta/inmanta-core/pull/7765#discussion_r1655064745